habitat_baselines/rl/ddppo/policy/running_mean_and_var.py

```diff
@@ -20,29 +19,31 @@ def __init__(self, n_channels: int) -> None:
         self._mean: torch.Tensor = self._mean
         self._var: torch.Tensor = self._var
         self._count: torch.Tensor = self._count
-        self._distributed = distrib.is_initialized()
 
     def forward(self, x: Tensor) -> Tensor:
         if self.training:
-            new_mean = F.adaptive_avg_pool2d(x, 1).sum(0, keepdim=True)  # type: ignore
-            new_count = torch.full_like(self._count, x.size(0))
+            n = x.size(0)
+            x_channels_first = (
+                x.transpose(1, 0).contiguous().view(x.size(1), -1)
+            )
+            new_mean = x_channels_first.mean(-1, keepdim=True)
+            new_count = torch.full_like(self._count, n)
 
-            if self._distributed:
+            if distrib.is_initialized():
                 distrib.all_reduce(new_mean)
                 distrib.all_reduce(new_count)
+                new_mean /= distrib.get_world_size()
 
-            new_mean = new_mean / new_count
-
-            new_var = F.adaptive_avg_pool2d((x - new_mean).pow(2), 1).sum(  # type: ignore
```

Comment:
Why you decided to get rid of `adaptive_avg_pool2d`?

Reply:
Just .mean() is typically faster and with the channels format it isn't needed